source/common/stats/stats_impl.cc

```diff
@@ -150,9 +150,24 @@ bool TagExtractorImpl::extractTag(const std::string& stat_name, std::vector<Tag>
 }
 
 RawStatData* HeapRawStatDataAllocator::alloc(const std::string& name) {
-  // This must be zero-initialized
-  RawStatData* data = static_cast<RawStatData*>(::calloc(RawStatData::size(), 1));
-  data->initialize(name);
+  absl::string_view key = name;
+
+  if (key.size() > Stats::RawStatData::maxNameLength()) {
+    key.remove_suffix(key.size() - Stats::RawStatData::maxNameLength());
+    ENVOY_LOG_MISC(
+        warn,
+        "Statistic '{}' is too long with {} characters, it will be truncated to {} characters", key,
+        key.size(), Stats::RawStatData::maxNameLength());
+  }
+
+  auto ret = stats_set_.insert(std::pair<std::string, RawStatData*>(std::string(key), nullptr));
```

Comment:
I'd specify a typedef StringRawDataMap for the map type in the header, and then use StringRawDataMap::value_type here rather than spelling out the pair.